app/models/llms/basic_workflow/cli.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+import argparse
+import sys
+from typing import Optional
+
+from app.config.logging_config import get_logger
+from .integration import ItineraryGenerator, save_to_session_history, save_conversation_to_file
+
+logger = get_logger("voyager_t800.cli")
+
+
+class VoyagerCLI:
+    def __init__(self, args):
+        self.generator = ItineraryGenerator()
+        self.session_history = []
+        self.args = args
+    
+    def _save_and_exit(self):
+        if self.session_history:
+            print("💾 Saving conversation before exit...")
+            save_conversation_to_file(self.session_history)
+        print("👋 Thank you for using Voyager T800!")
+
+    def _exit_without_saving(self):
+        print("👋 Thank you for using Voyager T800!")
+
+    def _exit(self):
+        if not self.args.not_save_history:
+            self._save_and_exit()
+        else:
+            self._exit_without_saving()
+
+    
+    def interactive_mode(self):
+        print("🎯 INTERACTIVE MODE")
+        print("Enter your travel request ('quit' to exit; 'help' to see commands):")
+        
+        while True:
+            try:
+                user_input = ''
+                if self.args.request:
+                    user_input = self.args.request
+                else:
+                    user_input = input("\n🌍 Your travel request: ").strip()
+                
+                if user_input.lower() in ['quit', 'exit', 'q']:
+                    self._exit()
+                    break
+                
+                if user_input.lower() == 'help':
+                    self._show_help()
+                    continue
+                
+                if user_input.lower() == 'history':
+                    self._show_history()
+                    continue
+                
+                if user_input.lower() == 'save' and not self.args.not_save_history:
+                    self._save_conversation()
+                    continue
+                elif user_input.lower() == 'save' and self.args.not_save_history:
+                    print("❌ Conversation history will not be saved.")
+                    continue
+                
+                if not user_input:
+                    print("❌ Please enter a valid travel request.")
+                    continue
+                
+                self._process_request(user_input)
+                
+            except Exception:
+                self._exit()
+                break
```

Comment:
1. Consider extracting the command parsing logic into its own method for improved readability and testability.
2. Add input validation for `user_input`, especially for edge cases and empty strings.